flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/nfa/SharedBuffer.java

```diff
@@ -202,7 +203,8 @@ public void prune(long pruningTimestamp) {
 		final K key,
 		final V value,
 		final long timestamp,
-		final DeweyNumber version) {
+		final DeweyNumber version,
+		final MatchingBehaviour matchingBehaviour) {
```

Comment:
I think the `SharedBuffer` should nothing know about the `MatchingBehaviour`. The `MatchingBehaviour` is something specific to the `NFA` and, thus, the `NFA` should be responsible for implementing it. This will give a better separation of concerns.